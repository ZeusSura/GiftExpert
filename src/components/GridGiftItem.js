import React from "react";
import PropTypes from 'prop-types'
const GridGiftItem = ({ imagen }) => {
  const { url, title } = imagen;
  return (
    <div className="animate__animated animate__rotateIn animate__slower">
      <div className="card">
        <img src={url} alt={title} />
        <p>{title}</p>
      </div>
    </div>
  );
};
GridGiftItem.propTypes={
   imagen:PropTypes.object.isRequired
}
export default GridGiftItem;
