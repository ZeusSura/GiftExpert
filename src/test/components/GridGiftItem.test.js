import { shallow } from "enzyme";
import React from "react";
import GridGiftItem from "../../components/GridGiftItem";

const img = {
  id: "CXijLuAwu955u",
  title: "windows microsoft GIF",
  url:"https://media3.giphy.com/media/CXijLuAwu955u/giphyâ€¦6kzxdnqldrd3k3psn84dxmkxjp3d4mqi8ky&rid=giphy.gif",
};
const wraper = shallow(<GridGiftItem imagen={img} />);
describe("Pruebas item grid", () => {
  test("Prueba de snapshot esperado", () => {
    expect(wraper).toMatchSnapshot();
  });
  test ('Pruebas sobre la url ',()=>{
    const p  =wraper.find('img')
    const attr = p.props();
    
    expect(attr.src.trim()).toBe(img.url)
  })

  test('Pruebas sobre el titulo',()=>{
    const p = wraper.find('p')
    const attr = p.text()
    expect(attr.trim()).toBe(img.title)
  })

});
