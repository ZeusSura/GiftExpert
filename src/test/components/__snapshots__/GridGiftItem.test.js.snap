// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas item grid Prueba de snapshot esperado 1`] = `
<div
  className="animate__animated animate__rotateIn animate__slower"
>
  <div
    className="card"
  >
    <img
      alt="windows microsoft GIF"
      src="https://media3.giphy.com/media/CXijLuAwu955u/giphyâ€¦6kzxdnqldrd3k3psn84dxmkxjp3d4mqi8ky&rid=giphy.gif"
    />
    <p>
      windows microsoft GIF
    </p>
  </div>
</div>
`;
